storage/rocksdb/ha_rocksdb.cc

```diff
@@ -8260,16 +8302,37 @@ int ha_rocksdb::update_sk(const TABLE *const table_arg, const Rdb_key_def &kd,
 
   const bool store_row_debug_checksums = should_store_row_debug_checksums();
 
+  m_sk_tails.clear();
+  // Reserve space for index flag fields
+  if (kd.m_total_index_flags_length > 0) {
+    m_sk_tails.allocate(0, kd.m_total_index_flags_length);
+  }
+
+  // Insert TTL timestamp
+  if (kd.has_ttl()) {
+    kd.write_index_flag_field(&m_sk_tails,
+                              reinterpret_cast<uchar *>(m_ttl_bytes),
+                              Rdb_key_def::TTL_FLAG);
+  }
+
   new_packed_size = kd.pack_record(
       table_arg, m_pack_buffer, row_info.new_data, m_sk_packed_tuple,
       &m_sk_tails, store_row_debug_checksums, row_info.hidden_pk_id);
 
   if (row_info.old_data != nullptr) {
+    m_sk_tails_old.clear();
     // The old value
     old_packed_size = kd.pack_record(
         table_arg, m_pack_buffer, row_info.old_data, m_sk_packed_tuple_old,
         &m_sk_tails_old, store_row_debug_checksums, row_info.hidden_pk_id);
 
+    const uchar *old_unpack_info = m_sk_tails_old.ptr();
```

Comment:
If pack_record() handled recording the ttl value, then this manipulation to adjust the sizes are all taken care of automatically.